### PR TITLE
feat(Radio): changes color and adds hover behavior

### PR DESCRIPTION
**Pull request by borisonr**
### Overview

changes the color of radio buttons to purple
adds a purple border (box shadow) on hover for unchecked buttons
makes the purple border (box shadow) thicker on hover for checked buttons

### PR Checklist

- [x] Related to designs: https://drive.google.com/file/d/1cZcB9osjWg0RoNRknZxqWkwJ1aUTZHVv/view and https://www.figma.com/file/xaosTDGWkXtTJBLPDV7ME2/Checkout---Clarity-improvments?node-id=389%3A4
- [x] Related to JIRA ticket: [GROW-2105(https://codecademy.atlassian.net/browse/GROW-2105)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602d6e2dbf1ccc45ea571e91--gamut-preview.netlify.app>https://602d6e2dbf1ccc45ea571e91--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602d6e2dbf1ccc45ea571e91>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.45-alpha.00eb4e.0<br/>@codecademy/gamut-labs@10.0.12-alpha.00eb4e.0<br/>@codecademy/gamut@23.2.2-alpha.00eb4e.0<br/>@codecademy/styleguide@27.2.2-alpha.00eb4e.0<br/></pre>

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Screenshot of a disabled radio with the outline

Going off of [the preview story](https://602d6e2dbf1ccc45ea571e91--gamut-preview.netlify.app/storybook/?path=/docs/atoms-forminputs-radio--radio) I think you want a `:not(:disabled)` in there.

File: packages/gamut/src/Form/styles/Radio.module.scss [hunk 1]
```diff
@@ -33,16 +33,21 @@
 }
 
 .radioInput:checked + .radioLabel::before {
-  background-color: $color-blue-500;
-  border: 0.25rem solid $color-white;
+  background-color: $color-hyper;
+  border: px-rem(5px) solid $color-white;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Form/styles/Radio.module.scss:37]: This a pretty aggressive outline - is 2px or 3px more accurate to the designs?

Also, the animation in the designs has a `transition`, could you add that?

**Comment by borisonr** [comment]
wait shoot i didnt realize it automatically adds reviewers, was not ready for review 😬

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.45-alpha.875cc1.0<br/>@codecademy/gamut-labs@10.0.12-alpha.875cc1.0<br/>@codecademy/gamut@23.2.2-alpha.875cc1.0<br/>@codecademy/styleguide@27.2.2-alpha.875cc1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602d8a214a42904df3cc8c66--gamut-preview.netlify.app>https://602d8a214a42904df3cc8c66--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602d8a214a42904df3cc8c66>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.45-alpha.ebdcd9.0<br/>@codecademy/gamut-labs@10.0.12-alpha.ebdcd9.0<br/>@codecademy/gamut@23.2.2-alpha.ebdcd9.0<br/>@codecademy/styleguide@27.2.2-alpha.ebdcd9.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.45-alpha.084b8d.0<br/>@codecademy/gamut-labs@10.0.12-alpha.084b8d.0<br/>@codecademy/gamut@23.2.2-alpha.084b8d.0<br/>@codecademy/styleguide@27.2.2-alpha.084b8d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602daec6edaee90fe4966006--gamut-preview.netlify.app>https://602daec6edaee90fe4966006--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602daec6edaee90fe4966006>Deploy Logs</a></p>

**Comment by borisonr** [comment]
okay actually ready for review now!

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM! Always fun to work on CSS animations. 😄 

I found it annoying to have to refresh the page to trigger the animation again... if you want ✨  bonus points ✨   you could add another option or two to the story.

**Comment by borisonr** [comment]
> LGTM! Always fun to work on CSS animations. 😄
> 
> I found it annoying to have to refresh the page to trigger the animation again... if you want ✨ bonus points ✨ you could add another option or two to the story.

i was about to do that but realized we have that in the radio group story

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.45-alpha.4fe8a3.0<br/>@codecademy/gamut-labs@10.0.12-alpha.4fe8a3.0<br/>@codecademy/gamut@24.0.0-alpha.4fe8a3.0<br/>@codecademy/styleguide@28.0.0-alpha.4fe8a3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602ea783d2de3516b1cdf7e9--gamut-preview.netlify.app>https://602ea783d2de3516b1cdf7e9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602ea783d2de3516b1cdf7e9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.46-alpha.4d5a61.0<br/>@codecademy/gamut-labs@10.0.13-alpha.4d5a61.0<br/>@codecademy/gamut@24.0.1-alpha.4d5a61.0<br/>@codecademy/styleguide@28.0.1-alpha.4d5a61.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602ea9dc16ddfe5317640615--gamut-preview.netlify.app>https://602ea9dc16ddfe5317640615--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602ea9dc16ddfe5317640615>Deploy Logs</a></p>

File: packages/gamut/src/Form/styles/Radio.module.scss [hunk 2]
```diff
@@ -15,38 +15,50 @@
...
+  transform: scale(1);
+  border: px-rem(5px) solid $color-white;
+}
+
+.radioInput:checked + .radioLabel::before {
+  box-shadow: 0 0 0 1px $color-hyper;
+}
+
+.radioInput:hover:not(:disabled) + .radioLabel::before {
+  box-shadow: 0 0 0 2px $color-hyper;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Form/styles/Radio.module.scss:55]: Oh, that checked-to-unchecked animation has a jarring size increase:

https://user-images.githubusercontent.com/3335181/108405634-4db8a100-71ef-11eb-805a-7a193b1203ae.mov

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Ah, a thing

File: packages/gamut/src/Form/styles/Radio.module.scss (hunk 2 above)
Comment (borisonr) [reply packages/gamut/src/Form/styles/Radio.module.scss:55]: 😢

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.46-alpha.86c3c6.0<br/>@codecademy/gamut-labs@10.0.13-alpha.86c3c6.0<br/>@codecademy/gamut@24.0.1-alpha.86c3c6.0<br/>@codecademy/styleguide@28.0.1-alpha.86c3c6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602eb7e8ddb4bb0da83b1cf8--gamut-preview.netlify.app>https://602eb7e8ddb4bb0da83b1cf8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602eb7e8ddb4bb0da83b1cf8>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
💯

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.46-alpha.88dcca.0<br/>@codecademy/gamut-labs@10.0.13-alpha.88dcca.0<br/>@codecademy/gamut@24.0.1-alpha.88dcca.0<br/>@codecademy/styleguide@28.0.1-alpha.88dcca.0<br/>@codecademy/tracking@0.5.0-alpha.88dcca.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602eba3dddb4bb145c3b1821--gamut-preview.netlify.app>https://602eba3dddb4bb145c3b1821--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602eba3dddb4bb145c3b1821>Deploy Logs</a></p>

**Comment by zippyzow** [comment]
Wow this looks so good compared to the current radio buttons! Question: Will there always be a default chosen so we will never have a true error state hence why the error looks the same as a regular option? By true error state I mean something like this: 
Screen Shot 2021-02-18 at 6 12 27 PM

**Review by zippyzow — APPROVED** [review]
👏

**Comment by borisonr** [comment]
> Wow this looks so good compared to the current radio buttons! Question: Will there always be a default chosen so we will never have a true error state hence why the error looks the same as a regular option? By true error state I mean something like this:
> Screen Shot 2021-02-18 at 6 12 27 PM

great question, @zippyzow 
for checkout there is always a default, not sure about the other instances though, but i will raise with tamar

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.47-alpha.43cc85.0<br/>@codecademy/gamut-labs@10.1.1-alpha.43cc85.0<br/>@codecademy/gamut@25.0.1-alpha.43cc85.0<br/>@codecademy/styleguide@29.1.1-alpha.43cc85.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602fe5eab3b0d60094c9cda2--gamut-preview.netlify.app>https://602fe5eab3b0d60094c9cda2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602fe5eab3b0d60094c9cda2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.47-alpha.05d127.0<br/>@codecademy/gamut-labs@10.1.1-alpha.05d127.0<br/>@codecademy/gamut@25.0.1-alpha.05d127.0<br/>@codecademy/styleguide@29.1.1-alpha.05d127.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602ffda4859d570cc323bd04--gamut-preview.netlify.app>https://602ffda4859d570cc323bd04--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602ffda4859d570cc323bd04>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.3-alpha.fdd314.0<br/>@codecademy/gamut@25.1.1-alpha.fdd314.0<br/>@codecademy/styleguide@29.2.1-alpha.fdd314.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6033ec7bf7eaa50ae051a1d9--gamut-preview.netlify.app>https://6033ec7bf7eaa50ae051a1d9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6033ec7bf7eaa50ae051a1d9>Deploy Logs</a></p>

**Pull request merged**